docs/src/references/agent/operation-workflow.md

```diff
@@ -328,6 +332,67 @@ Note that:
+
+- For instance, `input.url = "http://127.0.0.1:8000/tedge/file-transfer/foo.txt"`
+  adds an `url` property to the init message of the sub-command.
+- Similarly, `input.url = "${.payload.config_update.url}"`
+  adds an `url` property to the init message of the sub-command, extracting this url from the calling command state.
+- Several such arguments can be added to the sub-command initial state.
+  - All these individual arguments are combined into a JSON object value published on the sub-command MQTT topic.
+  - The calling command excerpts are injected into this initial state are defined using the same path conventions as for script handlers.
+  - If an `input.status` value is provided, this value will be ignored and replaced by `"init"`
+    as this is the only valid value for the initial state of a command.
+  - If no value at all is provided for `input`, then the default value is `{"status" = "init"}`.
+- For cases where there no one-to-one relationship between the properties of the current command state
+  and the properties of sub-command init state, the sub-command init state can be computed using a script.
+  - `input_script = "/some/script/which/stdout/is/used/as/sub-command/init/state"`
+  - The input script can be passed parameters extracted from the current command state,
+    *e.g.* `input_script = "/bin/extra_configuration_updates.sh ${.payload.device_profile}"`
+  - When the input script is successful, its stdout is used to determine the init state of the sub-command
+      - From this output, only the excerpt between a `:::begin-tedge:::` header and a `:::end-tedge:::` trailer is decoded.
+      - If this excerpt is a json payload, this payload is injected into the previous message payload
+        (adding new fields, overriding overlapping ones, keeping previous unchanged ones).
+      - If this excerpt is a json payload with a `status` field, then this status is ignored and its value replaced by `"init"`.
+  - When the input script fails, no subcommand is created and the command moves to the `failed` state.
+  - If both a script and explicit input properties are provided, the script output is applied first and the explicit properties second.
+- There is no way to assign a command identifier for the new sub-command instance.
+  - This identifier is generated by %%te%%, which creates, monitors and clears the MQTT topic for the sub-command.
+
+From an execution perspective, triggering a sub-command is similar to running a background script.
+
+- The calling command workflow is paused while the sub-command is processed.
+- The calling workflow move to an `on_exec` state from where to sub-command execution is watched till success, failure or timeout.
+- A specific `"await-command-completion"` action is to be used for this `on_exec` state where the sub-command completion is awaited.
+- The steps following the sub-command completion are defined on the `on_exec` state with `on_success`, `on_error` and `on_timeout` handlers.
```

Comment:
> question: why is sub-command similar to a background script?

In both case, the main workflow is moved into an intermediate state where the sub-command has been launched but not finalized yet". By contrast, with a regular script the main workflow is paused during the execution of the script, *but* stays in the same state. If for some reason the agent stops before the script ends, then this script will be relaunched again on agent restart. This is not be the case for sub-commands and back-ground scripts which are not relaunched if interrupted.

> By _calling command workflow is paused while the sub-command is processed_ I would assume it's more like a foreground script, where we await the result of the execution before continuing.

Yeah, *background* is perhaps not the best wording.

> After reading a bit more, I assume the reason is that workflows have to ultimately be resume-able (restart being a chief example, because after the restart we want to resume in a different state that the one which caused the restart, or else we'd just run the restart command again). Is my understanding correct?

Your understanding is correct. We need to persist the progress from "command to be launched" -> "command launched but not returned yet" -> "command terminated - successfully or not".